.github/workflows/cid-pullrequest.yml

```diff
@@ -0,0 +1,238 @@
+
+# name
+name: cid-pullrequest
+
+# triggers
+on:
+  workflow_dispatch:
+    inputs:
+      loglevel:
+        description: Log level
+        required: true
+        default: info
+        type: choice
+        options:
+          - debug
+          - info
+          - warn
+          - error
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - README.md
+      - LICENSE
+      - .github/**
+      - .gitignore
+      - .editorconfig
+      - renovate.json
+
+# permissions, see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions and https://docs.github.com/en/rest/overview/permissions-required-for-github-apps
+permissions:
+  actions: read # detection of GitHub Actions environment
+  checks: none
+  contents: read
+  deployments: none
+  id-token: none
+  issues: none
+  packages: none
+  pages: none
+  pull-requests: none
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+# cancel in progress when a new run starts
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+env:
+  CID_WORKFLOW: main
+  CID_VERSION: latest
+  CID_LOGLEVEL: ${{ github.event.inputs.loglevel || 'info' }}
+  # allowed modes are 'block' and 'audit'. Using https://github.com/step-security/harden-runner to harden the runner.
+  EGRESS_POLICY: block
+  # allowed endpoints for egress traffic if egress-policy is set to 'block'.
+  EGRESS_POLICY_ALLOWED_ENDPOINTS: >-
+    api.github.com:443
+    cdn01.quay.io:443
+    cdn02.quay.io:443
+    cdn03.quay.io:443
+    downloads.gradle.org:443
+    github.com:443
+    jcenter.bintray.com:443
+    kotlinlang.org:443
+    objects.githubusercontent.com:443
+    plugins-artifacts.gradle.org:443
+    plugins.gradle.org:443
+    quay.io:443
+    raw.githubusercontent.com:443
+    repo.maven.apache.org:443
+    repo1.maven.org:443
+    services.gradle.org:443
+    uploads.github.com:443
+  EGRESS_POLICY_ALLOWED_ENDPOINTS_BUILD: ""
+  EGRESS_POLICY_ALLOWED_ENDPOINTS_TEST: ""
+  EGRESS_POLICY_ALLOWED_ENDPOINTS_SCAN: >-
+    scanner.sonarcloud.io:443
+    semgrep.dev:443
+    sonarcloud.io:443
+  EGRESS_POLICY_ALLOWED_ENDPOINTS_PACKAGE: ""
+  EGRESS_POLICY_ALLOWED_ENDPOINTS_PUBLISH: >-
+    maven.pkg.github.com
+    oss.sonatype.org:443
+    s01.oss.sonatype.org:443
+
+# jobs
+jobs:
+  # info
+  info:
+    name: Info
+    runs-on: ubuntu-22.04 # https://github.com/actions/runner-images
+    timeout-minutes: 30
+    if: ${{ github.event.inputs.loglevel == 'debug' }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        with:
+          disable-sudo: true
+          egress-policy: ${{ env.EGRESS_POLICY }}
+          allowed-endpoints: ${{ env.EGRESS_POLICY_ALLOWED_ENDPOINTS }}
+      - name: prepare environment
+        uses: cidverse/ghact-cid-setup@main
```

Comment:
maybe the cid generator should pin the action version (`31e7177a4d98b05a05b4671f70df0ed199bfb9a1`)